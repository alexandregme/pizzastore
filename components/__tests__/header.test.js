import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';
import Header from '../header';
import TotalCost from "../totalcost";

describe("Head", ()=>{
    let mountedHeader;

    beforeEach(()=>{
        mountedHeader = shallow(<Header />);
    });

    it('renders without crashing', () => {
        shallow(<Header />);
        expect(mountedHeader).toMatchSnapshot();
    });

    it('expect to have header tag', () => {
        expect(mountedHeader.find('header')).toHaveLength(1);
    });

    it('expect to have nav tag', () => {
        expect(mountedHeader.find('nav')).toHaveLength(1);
    });

    it('expect to have one viewport meta tag', () => {
        expect(mountedHeader.find('Link[href="/"]')).toHaveLength(1);
    });

    it('expect to have one viewport meta tag', () => {
        expect(mountedHeader.find('Link[href="/cart"]')).toHaveLength(1);
    });

    it('expect to have TotalCost component', () => {
      expect(mountedHeader.find(TotalCost)).toHaveLength(1);
    });
});
