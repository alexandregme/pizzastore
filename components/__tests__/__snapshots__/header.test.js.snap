// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <header>
        <nav
          className="navbar navbar-expand-lg navbar-light bg-warning"
        >
          <div
            className="container"
          >
            <ul
              className=" nav navbar-nav flex-row"
            >
              <li
                className="nav-item"
              >
                <Link
                  href="/"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  href="/cart"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Cart
                  </a>
                </Link>
              </li>
              <li
                className="nav-item header-cart"
              >
                <Connect(SummaryCart)
                  className="nav-link navbar-brand"
                />
              </li>
            </ul>
          </div>
        </nav>
      </header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-expand-lg navbar-light bg-warning"
        >
          <div
            className="container"
          >
            <ul
              className=" nav navbar-nav flex-row"
            >
              <li
                className="nav-item"
              >
                <Link
                  href="/"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  href="/cart"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Cart
                  </a>
                </Link>
              </li>
              <li
                className="nav-item header-cart"
              >
                <Connect(SummaryCart)
                  className="nav-link navbar-brand"
                />
              </li>
            </ul>
          </div>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <ul
              className=" nav navbar-nav flex-row"
            >
              <li
                className="nav-item"
              >
                <Link
                  href="/"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  href="/cart"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Cart
                  </a>
                </Link>
              </li>
              <li
                className="nav-item header-cart"
              >
                <Connect(SummaryCart)
                  className="nav-link navbar-brand"
                />
              </li>
            </ul>
          </div>,
          "className": "navbar navbar-expand-lg navbar-light bg-warning",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className=" nav navbar-nav flex-row"
            >
              <li
                className="nav-item"
              >
                <Link
                  href="/"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Home
                  </a>
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  href="/cart"
                >
                  <a
                    className="nav-link navbar-brand"
                  >
                    Cart
                  </a>
                </Link>
              </li>
              <li
                className="nav-item header-cart"
              >
                <Connect(SummaryCart)
                  className="nav-link navbar-brand"
                />
              </li>
            </ul>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  <li
                    className="nav-item"
                  >
                    <Link
                      href="/"
                    >
                      <a
                        className="nav-link navbar-brand"
                      >
                        Home
                      </a>
                    </Link>
                  </li>,
                  <li
                    className="nav-item"
                  >
                    <Link
                      href="/cart"
                    >
                      <a
                        className="nav-link navbar-brand"
                      >
                        Cart
                      </a>
                    </Link>
                  </li>,
                ],
                <li
                  className="nav-item header-cart"
                >
                  <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />
                </li>,
              ],
              "className": " nav navbar-nav flex-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "/0",
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>,
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Home",
                      "className": "nav-link navbar-brand",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "/cart1",
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    href="/cart"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>,
                    "href": "/cart",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Cart",
                      "className": "nav-link navbar-brand",
                    },
                    "ref": null,
                    "rendered": "Cart",
                    "type": "a",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />,
                  "className": "nav-item header-cart",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "nav-link navbar-brand",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        "type": "nav",
      },
      "type": "header",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <header>
          <nav
            className="navbar navbar-expand-lg navbar-light bg-warning"
          >
            <div
              className="container"
            >
              <ul
                className=" nav navbar-nav flex-row"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    href="/"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    href="/cart"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item header-cart"
                >
                  <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />
                </li>
              </ul>
            </div>
          </nav>
        </header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <nav
            className="navbar navbar-expand-lg navbar-light bg-warning"
          >
            <div
              className="container"
            >
              <ul
                className=" nav navbar-nav flex-row"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    href="/"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    href="/cart"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item header-cart"
                >
                  <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />
                </li>
              </ul>
            </div>
          </nav>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <ul
                className=" nav navbar-nav flex-row"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    href="/"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    href="/cart"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item header-cart"
                >
                  <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />
                </li>
              </ul>
            </div>,
            "className": "navbar navbar-expand-lg navbar-light bg-warning",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className=" nav navbar-nav flex-row"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    href="/"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    href="/cart"
                  >
                    <a
                      className="nav-link navbar-brand"
                    >
                      Cart
                    </a>
                  </Link>
                </li>
                <li
                  className="nav-item header-cart"
                >
                  <Connect(SummaryCart)
                    className="nav-link navbar-brand"
                  />
                </li>
              </ul>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  Array [
                    <li
                      className="nav-item"
                    >
                      <Link
                        href="/"
                      >
                        <a
                          className="nav-link navbar-brand"
                        >
                          Home
                        </a>
                      </Link>
                    </li>,
                    <li
                      className="nav-item"
                    >
                      <Link
                        href="/cart"
                      >
                        <a
                          className="nav-link navbar-brand"
                        >
                          Cart
                        </a>
                      </Link>
                    </li>,
                  ],
                  <li
                    className="nav-item header-cart"
                  >
                    <Connect(SummaryCart)
                      className="nav-link navbar-brand"
                    />
                  </li>,
                ],
                "className": " nav navbar-nav flex-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "/0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      href="/"
                    >
                      <a
                        className="nav-link navbar-brand"
                      >
                        Home
                      </a>
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <a
                        className="nav-link navbar-brand"
                      >
                        Home
                      </a>,
                      "href": "/",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Home",
                        "className": "nav-link navbar-brand",
                      },
                      "ref": null,
                      "rendered": "Home",
                      "type": "a",
                    },
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "/cart1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      href="/cart"
                    >
                      <a
                        className="nav-link navbar-brand"
                      >
                        Cart
                      </a>
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <a
                        className="nav-link navbar-brand"
                      >
                        Cart
                      </a>,
                      "href": "/cart",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Cart",
                        "className": "nav-link navbar-brand",
                      },
                      "ref": null,
                      "rendered": "Cart",
                      "type": "a",
                    },
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(SummaryCart)
                      className="nav-link navbar-brand"
                    />,
                    "className": "nav-item header-cart",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "nav-link navbar-brand",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          "type": "nav",
        },
        "type": "header",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
